src/components/Lightbox.js

```diff
@@ -0,0 +1,233 @@
+
+    useEffect(() => {
+        if (!hasSiblingCarouselItems) {
+            return;
+        }
+
+        if (isActive) {
+            if (isImageLoaded && isFallbackVisible) {
+                // We delay hiding the fallback image while image transformer is still rendering
+                setTimeout(() => {
+                    setFallbackVisible(false);
+                    setFallbackLoaded(false);
+                }, 100);
+            }
+        } else {
+            if (isLightboxVisible && isLightboxLoaded) {
+                return;
+            }
+
+            // Show fallback when the image goes out of focus or when the image is loading
+            setFallbackVisible(true);
+        }
+    }, [hasSiblingCarouselItems, isActive, isImageLoaded, isFallbackVisible, isLightboxLoaded, isLightboxVisible]);
+
+    const fallbackSize = useMemo(() => {
+        if (!hasSiblingCarouselItems || (imageDimensions?.lightboxSize == null && imageDimensions?.fallbackSize == null) || containerSize.width === 0 || containerSize.height === 0) {
+            return;
+        }
+
+        const imageSize = imageDimensions.lightboxSize || imageDimensions.fallbackSize;
+
+        const {minScale} = getCanvasFitScale({canvasSize: containerSize, contentSize: imageSize});
+
+        return {
+            width: PixelRatio.roundToNearestPixel(imageSize.width * minScale),
+            height: PixelRatio.roundToNearestPixel(imageSize.height * minScale),
+        };
+    }, [containerSize, hasSiblingCarouselItems, imageDimensions]);
+
+    return (
+        <View
+            onPress={onPress}
+            style={[StyleSheet.absoluteFill, style]}
+            onLayout={updateCanvasSize}
+        >
+            {isContainerLoaded && (
+                <>
+                    {isLightboxVisible && (
+                        <View style={[StyleSheet.absoluteFill, {opacity: hasSiblingCarouselItems && isFallbackVisible ? 0 : 1}]}>
+                            <MultiGestureCanvas
+                                isActive={isActive}
+                                onScaleChanged={onScaleChanged}
+                                canvasSize={containerSize}
+                                contentSize={imageDimensions?.lightboxSize}
+                                zoomRange={zoomRange}
+                            >
+                                <Image
+                                    source={{uri: source}}
+                                    style={imageDimensions?.lightboxSize}
+                                    isAuthTokenRequired={isAuthTokenRequired}
+                                    onError={onError}
+                                    onLoadEnd={() => setImageLoaded(true)}
+                                    onLoad={(e) => {
+                                        const width = (e.nativeEvent?.width || 0) / PixelRatio.get();
+                                        const height = (e.nativeEvent?.height || 0) / PixelRatio.get();
```

Comment:
@chrispader Can you tell me why this was divided by `PixelRatio.get()`?

Reply:
Maybe Expo Image already took care of this internally.

Reply:
Yes, this should indeed be `* PixelRatio.get()` instead.

Reply:
Also, it should be added to the fallback image's `onLoad` function. Please make sure to add it there as well.